pkg/sql/internal.go

```diff
@@ -106,6 +117,47 @@ func MakeInternalExecutor(
 	}
 }
 
+// MakeInternalExecutorWithTxn creates an Internal Executor with txn related
+// information.
+func MakeInternalExecutorWithTxn(
```

Comment:
Can you add more commentary here about when this is going to get used and how it should be used and generally make it such that if somebody unfamiliar with all the accumulated history walked up to this function, they could figure out what's going on with it?

Reply:
Done